### ISSUE TITLE
Solve SonarCloud rule "Results of ~ and << operations on operands of underlying types unsigned char..."

### ISSUE THREAD

**Issue by mikir** [ndsev/zserio#622]
Enable the SonarCloud rule "[Results of ~ and << operations on operands of underlying types unsigned char and unsigned short should immediately be cast to the operand's underlying type](https://sonarcloud.io/organizations/ndsev/rules?open=cpp%3AS853&rule_key=cpp%3AS853)" and solve all found issues.

We used this in the generated code for bitmasks if bitmask type is e.g. `uint8_t`:

```
inline Bitmask operator~(Bitmask::Values lhs)
{
    // Add an explicit cast to the result of the "~" operator.
    return Bitmask(static_cast<Bitmask::underlying_type>(~static_cast<Bitmask::underlying_type>(lhs) & UINT8_C(63)));
}
```

**Comment by Mi-La** [comment]
Currently we are able to fix this for found issues since all or most of them are in Bitmasks. But in general this is related to https://github.com/ndsev/zserio/issues/86.

**Comment by Mi-La** [comment]
Fixes also sonar issues reported in #623.
